admin/api/routes.go

```diff
@@ -34,37 +34,68 @@ func (h *RoutesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	hostFilter := getFilterParameter(r, "host")
+	pathFilter := getFilterParameter(r, "path")
+	tagFilter := getFilterParameter(r, "tag")
+
 	var hosts []string
-	for host := range t {
-		hosts = append(hosts, host)
+
+	if hostFilter != nil {
+		hosts = append(hosts, *hostFilter)
+	} else {
+		for host := range t {
+			hosts = append(hosts, host)
+		}
+		sort.Strings(hosts)
 	}
-	sort.Strings(hosts)
 
 	var routes []apiRoute
+
 	for _, host := range hosts {
 		for _, tr := range t[host] {
-			for _, tg := range tr.Targets {
-				var opts []string
-				for k, v := range tg.Opts {
-					opts = append(opts, k+"="+v)
-				}
+			if pathFilter == nil || tr.Path == *pathFilter {
```

Comment:
What do you think about moving this check into a helper method? like pathMatchFilter? Or a generic match method can be used by both pathFilter and tags?

Reply:
Held off on a generic method since the path check is simple and the tag check requires looping through the tags.

Reply:
instead of passing path (string) into the function, you can pass paths ([]string). That way, you can use a generic function for both check. Though I don't have strong preference on this, so up to you. :)

Reply:
Good point. I lean towards keeping them separate also because we may wish to later modify or expand the tag filter logic, for instance to check for multiple tags and return true only if all are present.